pool/entities.go

```diff
@@ -0,0 +1,256 @@
+package pool
+
+type ID string
+
+type Name string
+
+// I don't know i have to bring Peer Available storage and other metric inside Peer
+type Peer string
+
+type ReplicationFactor int8
+
+// Requirement of joining the pool
+type Requirement struct {
+	Latency   string `json:"latency"`
+	Bandwidth string `json:"bandwidth"`
+}
+
+// ResourceRequest object (it is much more complicated for simplicity we think as file).
+type ResourceRequest struct {
+	ID  ID `json:"id"`
+	FileName  Name `json:"filename"`
+	Owner     Peer  `json:"owner"`
+	Size      string `json:"size"`
+	ReplicationFactor ReplicationFactor `json:"rp"`
+}
+
+type Resource struct{
+	ResourceRequest
```

Comment:
Why `Resource` is also a `ResourceRequest`?

Reply:
I need to store the ResourceRequest as an event. (the allocation can be calculated on the state).
And I need the same Fileds for the pool state so this is my solution.